core/trino-main/src/main/java/io/trino/operator/DirectExchangeClient.java

```diff
@@ -291,23 +292,22 @@ private boolean addPages(HttpPageBufferClient client, List<Slice> pages)
         checkState(!completedClients.contains(client), "client is already marked as completed");
         // Compute stats before acquiring the lock
         long responseSize = 0;
-        for (Slice page : pages) {
-            responseSize += page.length();
+        if (!pages.isEmpty()) {
+            for (Slice page : pages) {
+                responseSize += page.length();
+            }
+            buffer.addPages(client.getRemoteTaskId(), pages);
```

Comment:
add a comment that buffer can potentially be closed here but it is not a problem. Assuming it is not :)